templates/tab/js/default/src/components/sample/Graph.jsx

```diff
@@ -1,12 +1,23 @@
-import React from "react";
-import { Button } from "@fluentui/react-northstar";
+import "./Graph.css";
 import { useGraph } from "./lib/useGraph";
-import { ProfileCard } from "./ProfileCard";
+import { Providers, ProviderState } from '@microsoft/mgt-element';
+import { TeamsFxProvider } from '@microsoft/mgt-teamsfx-provider';
+import { Button } from "@fluentui/react-northstar";
+import { Design } from './Design';
+import { PersonCardFluentUI } from './PersonCardFluentUI';
+import { PersonCardGraphToolkit } from './PersonCardGraphToolkit';
 
 export function Graph() {
   const { loading, error, data, reload } = useGraph(
-    async (graph) => {
+    async (graph, credential, scope) => {
+      // Call graph api directly to get user profile information
       const profile = await graph.api("/me").get();
+
+      // Initialize Graph Toolkit TeamsFx provider
+      const provider = new TeamsFxProvider(credential, scope);
+      Providers.globalProvider = provider;
+      Providers.globalProvider.setState(ProviderState.SignedIn);
```

Comment:
If we always need to initialize this provider, maybe we can move the initialization into useGraph hook.
```
useGraph(async (graph) => { // business logics});


export function useGraph<T>(asyncFunc: (graph) => Promise<T>) {
   ......
   const provider = new TeamsFxProvider(credential, scope);
   Providers.globalProvider = provider;
   Providers.globalProvider.setState(ProviderState.SignedIn);
   return await asyncFunc(graph);
}
```

Reply:
The main purpose to put these logic here is to demonstrate and compare the graph api and graph toolkit calls in the same place.

Another reason is that if we put it into useGraph function, when user click authentication button for the first time, it will not call `setState(ProviderState.SignedIn)`, and the graph toolkit person card will not show, user need to manually refresh to page to make it work.